### ISSUE TITLE
failed builds leave bad state

### ISSUE THREAD

**Issue by dcgibbons** [kerl/kerl#202]
If a build fails for some reason, you can't rebuild unless you delete the build.

Example: I forgot to install ncurses-devel on linux, did a build... failed... installed ncurses-devel, tried the build command again, and received:
```
[cgibbons@eyeball ~]$ kerl build 19.3 erlang_r19_3
Verifying archive checksum...
Checksum verified (a8c259ec47bf84e77510673e1b76b6db)
Building Erlang/OTP 19.3 (erlang_r19_3), please wait...
Configure options have changed. Reconfiguring...
Configure failed.
checking for tgetent in -ltermcap... no
checking for tgetent in -ltermlib... no
configure: error: No curses library functions found
configure: error: /bin/sh '/home/cgibbons/.kerl/builds/erlang_r19_3/otp_src_19.3/erts/configure' failed for erts
/home/cgibbons/.kerl/builds/erlang_r19_3/otp_src_19.3/configure  --cache-file=/dev/null
./otp_build: line 333: ./configure: No such file or directory
Makefile:248: /home/cgibbons/.kerl/builds/erlang_r19_3/otp_src_19.3/make/x86_64-unknown-linux-gnu/otp_ded.mk: No such file or directory
make: *** No rule to make target `/home/cgibbons/.kerl/builds/erlang_r19_3/otp_src_19.3/make/x86_64-unknown-linux-gnu/otp_ded.mk'.  Stop.
/home/cgibbons/.kerl/builds/erlang_r19_3/otp_src_19.3/configure  --cache-file=/dev/null
./otp_build: line 333: ./configure: No such file or directory

Please see /home/cgibbons/.kerl/builds/erlang_r19_3/otp_build_19.3.log for full details.
```

**Comment by dcgibbons** [comment]
Once I remove the build directory with ```rm -fr ~/.kerl/builds/erlang_r19_3``` then I the build can proceed.

**Comment by mrallen1** [comment]
Thanks for the ticket. The line `Configure options have changed. Reconfiguring...` is intended to address this problem and clearly failed to do so. So maybe we need to look at its implementation to find a more robust solution.

**Comment by paulo-ferraz-oliveira** [comment]
Could the solution to this simply be to remove the incomplete build files from `kerl`'s cache?

**Comment by jadeallenx** [comment]
Yes that's likely a better way to do this - I think at the time I was trying to be more strategic.

**Comment by paulo-ferraz-oliveira** [comment]
Here's what I did (a lot changes in 6 years but I wanted to confirm the issue was gone)...

1. started with a clean Ubuntu container
2. `apt-get -y update`
3. `apt-get -y install curl git`
4. `git clone https://github.com/kerl/kerl`
4.1. no env. variables set that affect `kerl`'s config.
5. `./kerl update releases`  # oldest supported is 24.3.4.14
6. `./kerl build 24.3.4.14`

Got:

```console
...
Auto cleaning all artifacts except the log file...
(use KERL_AUTOCLEAN=0 to keep build on failure, if desired)
Cleaning up compilation products for '24.3.4.14' under /root/.kerl/builds...
... done.
```

Re-ran `./kerl build 24.3.4.14`.

Got same result.

The latter seems to indicate `KERL_AUTOCLEAN` is doing what it's supposed to, by default - Enabled.

I also listed the corresponding build `builds`' subfolder, and the shell message is truthful:

```console
~/.kerl/builds/24.3.4.14# ls -al
~/.kerl/builds/24.3.4.14# ls -al
total 12
drwxr-xr-x 2 root root 4096 Oct 26 21:21 .
drwxr-xr-x 3 root root 4096 Oct 26 21:21 ..
-rw------- 1 root root 2038 Oct 26 21:21 otp_build_24.3.4.14.log
```

Closing as nothing indicates this issue is still present.
